Fix metadata user data parse

superzc-github opened the pull request:
### What does this PR do?
This is to make sure user-data in grains metadata will be returned correctly
### What issues does this PR fix or reference?
N/A

### Previous Behavior
user-data content will be cut by this current condition:
elif "=" in line:
key, value = line.split("=")
ret[value] = _search(prefix=os.path.join(prefix, key))

### New Behavior
user-data will be return with its whole content directly with any further cut/parse for it's content

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

Commit:
fix(salt/grains/metadata.py): fix user-data content cut by = issue in grains meatadata parsing

Commit:
add change log

Commit:
remove wrong change log

Commit:
add change log

dwoz commented:
@superzc-github Are you able to add a test case for this change?

twangboy reviewed (changes requested):
Please create a test for this. Also missing a blankline in the changelog

Commit:
Add regression test + tighten user-data predicate (#62061)

Tighten the new branch in salt/grains/metadata.py::_search() to fire
only on `line == "user-data"` instead of the original
`line.endswith(("user-data"))` (which was a string-in-parens, not a
tuple, and matched any sub-path line ending in "user-data" — e.g. a
`security-credentials/` alias of the form `something=user-data`).

Add tests/pytests/unit/grains/test_metadata.py with three cases:
  - user-data with `=` in body is returned verbatim
  - missing user-data returns None without crashing
  - other `key=value` metadata leaves still parse via the `=`-splitter

Rewrite changelog/62061.fixed as coherent prose and add the missing
trailing newline.

Co-authored-by: SuperZC <superzc-github@users.noreply.github.com>
